packages/twenty-front/src/modules/object-record/record-board/record-board-column/hooks/useAddNewOpportunity.ts

```diff
@@ -47,11 +47,19 @@ export const useAddNewOpportunity = (position: string) => {
   );
 
   const handleAddNewOpportunityClick = useCallback(() => {
-    setIsCreatingCard(true);
+    if (isDefined(isOpportunitiesCompanyFieldDisabled)) {
+      handleEntitySelect();
+    } else {
+      setIsCreatingCard(true);
+    }
```

Comment:
the condition is wrong here
```suggestion
    if (isDefined(isOpportunitiesCompanyFieldDisabled)) {
      
      setIsCreatingCard(true);
    } else {
      handleEntitySelect();
    }
```

Reply:
If company field is disabled we directly create the card, otherwise we go through the process of picking a company.